core/src/main/java/com/coderanch/blackjack/MiniGame.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.blackjack;
+
+import com.coderanch.util.cli.InputUtility;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.security.SecureRandom;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Locale;
+import java.util.function.Predicate;
+
+import static com.coderanch.util.require.Require.requireThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * Mini game of Blackjack.
+ */
+public final class MiniGame {
+
+    /**
+     * Highest score in Blackjack.
+     */
+    private static final int MAX_LEGAL_SCORE = 21;
+
+    /**
+     * The predicate used for validating user input.
+     */
+    private static final Predicate<? super String> PLAY_PREDICATE = InputUtility.oneOfTheseIgnoringCase("hit", "pass");
```

Comment:
I'd just inline this predicate in the line where it is used, because it reads more fluently:

`inputUtility.nextString("hit or pass?", oneOfTheseIgnoringCase("hit", "pass"))`

You can then remove this constant.